apps/src/lib/levelbuilder/lesson-editor/StandardsEditor.jsx

```diff
@@ -135,10 +145,70 @@ class StandardsEditor extends Component {
     this.handleRemoveStandardDialogClose();
   };
 
+  handleSelectFramework = e => {
+    const frameworkShortcode = e.target.value;
+    this.setState({frameworkShortcode});
+  };
+
+  constructStandardOption = standard => ({
+    value: standard.shortcode,
+    label: `${standard.frameworkShortcode.toUpperCase()} - ${
+      standard.shortcode
+    } - ${standard.description}`,
+    standard: standard
+  });
+
+  constructSearchOptions = json => {
+    const existingShortcodes = this.props.standards.map(
+      standard => standard.shortcode
+    );
+    const standards = json
+      // Filter any that are already added to lesson
+      .filter(standard => !existingShortcodes.includes(standard.shortcode))
+      .map(standard => this.constructStandardOption(standard));
+    return {options: standards};
+  };
+
+  onSearchSelect = option => {
+    this.props.addStandard(option.standard);
+  };
+
   render() {
     const columns = this.getColumns();
     return (
       <div>
+        <label>
+          <strong>Filter by framework</strong>
+        </label>
+        <select onChange={this.handleSelectFramework} style={styles.select}>
+          <option value="">(none)</option>
+          <option value="iste">ISTE Standards for Students</option>
+          <option value="ccela">
+            Common Core English Language Arts Standards
+          </option>
+          <option value="ccmath">Common Core Math Standards</option>
+          <option value="ngss">Next Generation Science Standards</option>
+          <option value="csta">
+            CSTA K-12 Computer Science Standards (2017)
+          </option>
+          <option value="csp2021">CSP Conceptual Framework</option>
+        </select>
+        <label>
+          <strong>Select a Standard to add</strong>
+        </label>
+        <SearchBox
+          // Specify a key in order to force this component to remount when
```

Comment:
Nice find!

Reply:
Thanks! I did notice one related problem, which is that the logic for filtering out standards which are already in the lesson does not get updated after you add a standard. so, if you add a standard, then search for the same thing again, the standard you just added will still be in the list. https://codedotorg.atlassian.net/browse/PLAT-886